### PR TITLE
SPUI-497/489/488 - Sign up flow

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/auth/screens/login/login.dart [hunk 1]
```diff
@@ -28,86 +20,149 @@ class Login extends HookWidget {
...
-                onChanged: (value) {
-                  credentialsN.updateAndValidatePassword(value);
-                },
-              ),
-              const SpaceH40(),
-              ForgotPasswordButton(
-                onTap: () => navigatorPush(context, const ForgotPassword()),
+          child: Expanded(
+            child: Material(
+              color: SColorsLight().grey5,
```
Comment (eli1stark) [inline lib/auth/screens/login/login.dart]: use sColorPod

File: lib/auth/screens/login/login.dart [hunk 2]
```diff
@@ -28,86 +20,149 @@ class Login extends HookWidget {
...
+                              onErrorIconTap: () {
+                                _showErrorNotification(notificationQueueN);
+                              },
+                              errorNotifier: emailError.value,
+                            ),
+                          ),
+                        ),
+                        const SDivider(),
+                        Material(
+                          color: SColorsLight().white,
```
Comment (eli1stark) [inline lib/auth/screens/login/login.dart]: sColorsPod

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 3]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
-                  header: 'Enter password',
-                  hintText: 'Enter password',
-                  autofocus: true,
-                  autofillHints: const [AutofillHints.password],
-                  onChanged: (value) {
-                    credentialsN.updateAndValidatePassword(value);
-                  },
+            child: Expanded(
+              child: Material(
+                color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:55]: sColorsPod

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 4]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Container(
+                      color: Colors.white,
+                      padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:61]: use SPaddingH24

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 5]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
-                    credentialsN.updateAndValidatePassword(value);
-                  },
+            child: Expanded(
+              child: Material(
+                color: Colors.grey[200],
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Container(
+                      color: Colors.white,
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:60]: sColorsPod

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 6]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
+                        labelText: 'Password',
+                        onErrorIconTap: () {
+                          _showErrorNotification(notificationQueueN);
+                        },
+                        errorNotifier: passwordError.value,
+                        autofocus: true,
+                      ),
+                    ),
+                    Padding(
+                      padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:76]: use SPaddingH24

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 7]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
+                      child: PasswordValidation(
+                        password: credentials.password,
+                      ),
+                    ),
+                    const Spacer(),
+                    Padding(
+                      padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:83]: use SPaddingH24

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 8]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
+                              operation: AuthOperation.register,
+                            );
+                          }
+                        },
+                      ),
+                    ),
+                    Container(
+                      width: double.infinity,
+                      height: 24.h,
+                      color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:101]: sColorsPod

File: lib/auth/screens/register/register.dart [hunk 9]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
     final credentialsN = useProvider(credentialsNotipod.notifier);
+    final notificationQueueN = useProvider(sNotificationQueueNotipod.notifier);
+    final emailError = useValueNotifier(StandardFieldErrorNotifier());
 
-    return PageFrame(
-      header: 'Create an account',
-      onBackButton: () => Navigator.pop(context),
+    return SPageFrame(
+      header: Padding(
+        padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:25]: SPageFrameWithPadding

**Review by eli1stark — CHANGES_REQUESTED** [review]

**Comment by eli1stark** [comment]
not finished

File: lib/auth/screens/register/register.dart [hunk 10]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
-            const SpaceH40(),
-            AppTextField(
-              header: 'Enter your email',
-              hintText: 'Email address',
-              autofocus: true,
-              autofillHints: const [AutofillHints.email],
-              onChanged: (value) => credentialsN.updateAndValidateEmail(value),
+        child: Expanded(
+          child: Material(
+            color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:34]: sColorsPod

File: lib/auth/screens/register/register.dart [hunk 11]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Container(
+                  color: Colors.white,
+                  padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:40]: SPaddingH24

File: lib/auth/screens/register/register.dart [hunk 12]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+                    },
+                    onErrorIconTap: () {
+                      _showErrorNotification(notificationQueueN);
+                    },
+                    errorNotifier: emailError.value,
+                  ),
+                ),
+                const Spacer(),
+                Container(
+                  color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:57]: sColorsPod

File: lib/auth/screens/register/register.dart [hunk 13]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+                  child: Padding(
+                    padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:59]: SPaddingH24

File: lib/auth/screens/register/register.dart [hunk 14]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+                      onCheckboxTap: () => credentialsN.checkPolicy(),
+                      onUserAgreementTap: () =>
+                          launchURL(context, userAgreementLink),
+                      onPrivacyPolicyTap: () =>
+                          launchURL(context, privacyPolicyLink),
+                    ),
+                  ),
+                ),
+                Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 24.w),
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:76]: SPaddingH24

File: lib/auth/screens/register/register.dart [hunk 15]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+                          _showErrorNotification(notificationQueueN);
+                        }
+                      }
+                    },
+                  ),
+                ),
+                Container(
+                  width: double.infinity,
+                  height: 24.h,
+                  color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:98]: sColorsPod

File: lib/auth/shared/components/password_validation/password_validation.dart [hunk 16]
```diff
@@ -21,32 +21,23 @@ class PasswordValidation extends StatelessWidget {
...
-        const SpaceH10(),
-        const Text(
-          'Your password must: ',
-          style: TextStyle(
-            color: Colors.grey,
-          ),
+        Container(
+          width: double.infinity,
+          height: 24.h,
+          color: Colors.grey[200],
```
Comment (eli1stark) [inline lib/auth/shared/components/password_validation/password_validation.dart:27]: sColorsPod

File: lib/auth/screens/register/register.dart [hunk 17]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
+                          );
+                        } else {
+                          emailError.value.enableError();
+                          _showErrorNotification(notificationQueueN);
+                        }
+                      }
+                    },
+                  ),
+                ),
+                Container(
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:95]: repeated widget 1

File: lib/auth/shared/components/password_validation/password_validation.dart [hunk 18]
```diff
@@ -21,32 +21,23 @@ class PasswordValidation extends StatelessWidget {
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        const SpaceH10(),
-        const Text(
-          'Your password must: ',
-          style: TextStyle(
-            color: Colors.grey,
-          ),
+        Container(
```
Comment (eli1stark) [inline lib/auth/shared/components/password_validation/password_validation.dart:24]: repeated widget 1

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 19]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
+                            authenticationN.authenticate(
+                              email: credentials.email,
+                              password: credentials.password,
+                              operation: AuthOperation.register,
+                            );
+                          }
+                        },
+                      ),
+                    ),
+                    Container(
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:98]: repeated widget 1

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 20]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
           credentialsN.updateAndValidatePassword('');
           return Future.value(true);
         },
-        child: PageFrame(
-          header: 'Create a password',
-          onBackButton: () {
-            Navigator.pop(context);
-            credentialsN.updateAndValidatePassword('');
-          },
+        child: SPageFrame(
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:44]: SPageFrameWithPadding

File: lib/auth/shared/notifiers/credentials_notifier/credentials_notifier.dart [hunk 21]
```diff
@@ -82,8 +82,8 @@ class CredentialsNotifier extends StateNotifier<CredentialsState> {
         state.readyToLogin;
   }
 
-  bool get emailValidAndPolicyChecked {
-    return state.emailValid && state.policyChecked;
+  bool get emailIsNotEmptyAndPolicyChecked {
```
Comment (eli1stark) [inline lib/auth/shared/notifiers/credentials_notifier/credentials_notifier.dart:85]: we need to check email using regex

File: lib/auth/screens/register/components/register_password_screen.dart [hunk 22]
```diff
@@ -43,53 +41,87 @@ class RegisterPasswordScreen extends HookWidget {
...
-              ],
+              ),
             ),
           ),
         ),
       ),
     );
   }
+
+  void _showErrorNotification(SNotificationQueueNotifier notifier) {
```
Comment (eli1stark) [inline lib/auth/screens/register/components/register_password_screen.dart:113]: repeated logic 2

File: lib/auth/screens/register/register.dart [hunk 23]
```diff
@@ -18,40 +17,106 @@ class Register extends HookWidget {
...
-              active: credentialsN.emailValidAndPolicyChecked,
-            )
-          ],
+          ),
         ),
       ),
     );
   }
+
+  void _showErrorNotification(SNotificationQueueNotifier notifier) {
```
Comment (eli1stark) [inline lib/auth/screens/register/register.dart:108]: repeated logic 2

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/auth/screens/login/login.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/auth/screens/login/login.dart]: done

File: lib/auth/screens/login/login.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/auth/screens/login/login.dart]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:55]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:61]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:60]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:76]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:83]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 8 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:101]: done

File: lib/auth/screens/register/register.dart (hunk 9 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:25]: done

File: lib/auth/screens/register/register.dart (hunk 10 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:34]: done

File: lib/auth/screens/register/register.dart (hunk 11 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:40]: done

File: lib/auth/screens/register/register.dart (hunk 12 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:57]: done

File: lib/auth/screens/register/register.dart (hunk 13 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:59]: done

File: lib/auth/screens/register/register.dart (hunk 14 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:76]: done

File: lib/auth/screens/register/register.dart (hunk 15 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:98]: done

File: lib/auth/shared/components/password_validation/password_validation.dart (hunk 16 above)
Comment (VovaMedvediev) [reply lib/auth/shared/components/password_validation/password_validation.dart:27]: done

File: lib/auth/screens/register/register.dart (hunk 17 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart:95]: done

File: lib/auth/shared/components/password_validation/password_validation.dart (hunk 18 above)
Comment (VovaMedvediev) [reply lib/auth/shared/components/password_validation/password_validation.dart:24]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 19 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:98]: done

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 20 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:44]: done

File: lib/auth/shared/notifiers/credentials_notifier/credentials_notifier.dart (hunk 21 above)
Comment (VovaMedvediev) [reply lib/auth/shared/notifiers/credentials_notifier/credentials_notifier.dart:85]: no, we are able to input any not-empty email. If the email is wrong the user will see error notification

File: lib/auth/screens/register/components/register_password_screen.dart (hunk 22 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/components/register_password_screen.dart:113]: done

File: lib/auth/screens/register/register.dart (hunk 23 above)
Comment (VovaMedvediev) [reply lib/auth/screens/register/register.dart]: done

File: lib/auth/shared/components/password_validation/password_validation.dart [hunk 24]
```diff
@@ -21,11 +21,7 @@ class PasswordValidation extends StatelessWidget {
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        Container(
-          width: double.infinity,
-          height: 24.h,
-          color: Colors.grey[200],
-        ),
+        const Grey24HPadding(),
```
Comment (eli1stark) [inline lib/auth/shared/components/password_validation/password_validation.dart:24]: Rename to GreyPaddingH24 to our app standards

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/auth/shared/components/password_validation/password_validation.dart (hunk 24 above)
Comment (VovaMedvediev) [reply lib/auth/shared/components/password_validation/password_validation.dart:24]: components is deleted

**Pull request merged**